### DISCUSSION TITLE
How to override local timezone when using date commands?

### DISCUSSION THREAD

**Discussion by ghost** [nushell/nushell#10790] [Q&A]
I have a use-case where I need to override the local timezone. I usually do it like so: `TZ=America/New_York ^date`. I can't seem to do it in nushell. I'd expect it to look something like the following: `with-env [TZ America/New_York] {date now}`.

**Comment by fdncred** [comment]
This is how we do that. `date now | date to-timezone America/New_York`

**Reply by ghost** [reply]
Ah, yes. I do know that command. Hmm. I think my use-case is a bit more specific than my current example, which is a bit more difficult to do in nushell. In the example below, I need a UTC timestamp of the overridden timezone. Think being in a UTC container, but needing to supply a local timezone to a server which only accepts UTC timestamps 🙈  Is this possible? The example below works for me, but I'd like to do it with just nushell.

```
TZ=America/New_York ^date -u
```

So, what I was thinking may look more like the following:
```
with-env [TZ America/New_York] {date now | date to-timezone "UTC"}
```

**Reply by ghost** [reply]
This is another option that I'm able to do, but once I'm in an interactive nushell, I can't change the local timezone. I'm stuck with it 😅 

```
TZ="America/New_York" nu -c 'date now'
```

**Reply by fdncred** [reply]
If you search the repo for `locale` you'll see that there is an env var named `NU_TEST_LOCALE_OVERRIDE` that changes some things. I don't think `date now` knows about it tough.

Your request assumes all date commands recognize the `TZ` env var, which I'm pretty sure they don't right now.

**Reply by ghost** [reply]
If I understand correctly, I think it may have to do with [chrono caching the timezone](https://github.com/chronotope/chrono/blob/a44142528eb8bab32f9e16cb74e84bb060f4a667/src/offset/local/unix.rs#L25-L29).

**Reply by fdncred** [reply]
I haven't looked into it but that seems plausible.
